ruby/lib/minitest/queue.rb

```diff
@@ -266,6 +266,11 @@ def run_from_queue(reporter, *)
       reopen_previous_step
       puts red("The heartbeat process died. This worker is exiting early.")
       exit!(41)
+    rescue StandardError => e
+      reopen_previous_step
+      puts red("This worker died because of a queue or framework error:")
+      puts red("#{e.class}: #{e.message}")
+      exit!(42)
```

Comment:
I think the point is to distinguish between "transcient queue errors" (e.g. can't reach the Redis queue), and application errors (e.g. there is a bug in a test that somehow escape Minitest).

As such the message is weird to me. Should be something like `This worker exited because of an uncaught application error.`